source/core_mqtt.c

```diff
@@ -719,8 +726,11 @@ static int32_t recvExact( const MQTTContext_t * pContext,
             totalBytesRecvd = bytesRecvd;
             receiveError = true;
         }
-        else
+        else if( bytesRecvd > 0 )
         {
+            /* Reset the starting time as we have received some data from the network. */
+            entryTimeMs = getTimeStampMs();
```

Comment:
This name is no longer accurate, something like `lastByteTimeMs` would be better (also since we know at least two bytes were read before this function was called, so that will also be the last byte timestamp at time of entry)